doc/nixbuild.net.md

```diff
@@ -0,0 +1,96 @@
+nixbuild.net access
+===================
+
+The Smart Contracts tribe has a [nixbuild.net](https://nixbuild.net/) account which can be used to delegate Nix builds[^system] to powerful autoscaled hardware, with shared build results for all users of the account.
+
+[^system]: As of this writing, only for Linux on x86 and ARM; up-to-date list of systems [here](https://docs.nixbuild.net/supported-systems/index.html).
+
+Getting access
+---------------
+
+Pending at least the [initial prototype](https://input-output.atlassian.net/browse/PLT-4253) of the tribe-wide access control system, initial access for a user is a manual process. Please reach out to `Shea Levy` on Slack for access.
+
+**Note that the given tokens are secrets and should be managed as such!**
+
+Initial tokens will be set to expire around 2023-04-19, with the hope that by then an automated token server will exist. Use `biscuit inspect` (see [Managing access tokens](#managing-access-tokens)) to confirm the exact expiration.
+
+
+Setting up remote builders
+-------------------------
+
+nixbuild.net has documentation for setting everything up on [NixOS](https://docs.nixbuild.net/getting-started/#quick-nixos-configuration), or for configuring [SSH](https://docs.nixbuild.net/getting-started/#ssh-configuration) and [Nix](https://docs.nixbuild.net/getting-started/#nix-configuration) separately. From there, you can find links to more detailed documentation on remote builders. However, due to our usage of biscuit tokens instead of SSH keys changes to the SSH configuration are needed.
+
+Instead of an entry like:
+
+```
+Host eu.nixbuild.net
+  PubkeyAcceptedKeyTypes ssh-ed25519
+  IdentityFile /path/to/your/private/key
+```
+
+Your entry should look like:
+
+```
+# when connecting to eu.nixbuild.net
+Host eu.nixbuild.net
+  # don't use any authentication mechanisms SSH expects (password, SSH keys, etc.)
+  PreferredAuthentications none
+  # log in as the 'authtoken' user
+  # The token itself contains user information that nixbuild.net needs
+  User authtoken
+  # Send the 'token' environment variable from Nix to nixbuild.net in the SSH session.
+  SendEnv token
+  # Alternatively to SendEnv, you can have a line like:
+  #   SetEnv token=<token>
+  # This will mean you don't have to modify Nix's environment (see below).
+  # But this will involve putting your token, which is a secret, directly into
+  # a configuration file.
+```
+
+and then you should ensure that the environment of your Nix daemon (or, if you're running in single-user mode, your Nix build command) has the `token` variable set to the contents of your token.
```

Comment:
Hmmm, is it possible to do this securely on NixOS? I guess I can import the token value from a file and not check in that file into my nix config.

Maybe we can request some doc on doing this from nixbuild.net.

Reply:
Here's what I did:
```
cat secrets >> .gitignore
touch secrets/mpj-io-biscuit-token
git add -f secrets
git commit -m "Add empty secrets so flakes are happy :("
vim secrets/mpj-io-biscuit-token
... add correct content
```
in the ssh config
```
  SetEnv token={builtins.readFile ./secrets/mpj-io-biscuit-token}
```

So it appears in my SSH config file locally, but that's it.

Reply:
I'm planning to do this using a service overlay that can live in `/etc/systemd/system/nix-daemon.service.d` and is just a symlink to some actual private file, I'll document it once I have it working. Upstream doc ticket opened
https://github.com/nixbuild/feedback/issues/15

Reply:
Documented this now https://github.com/input-output-hk/sc-dev-platform/blob/111b6f25dcb44bf41c840db0247f97acb1097705/doc/nixbuild.net.md#setting-up-remote-builders